gitk-git/gitk

```diff
@@ -415,7 +415,8 @@ proc start_rev_list {view} {
     if {[catch {
 	set fd [open [concat | git log --no-color -z --pretty=raw $show_notes \
 			--parents --boundary $args --stdin \
-			"<<[join [concat $revs "--" $files] "\\n"]"] r]
+			"<<[join [concat $revs "--" \
+				[string map {\\ \\\\} $files]] "\\n"]"] r]
```

Comment:
Does this work with file names containing spaces? Double quotes? Asterisks?

Reply:
I initially ignored spaces since I misinterpreted the resulting error message as a bug originating somewhere else. Taking a second look this can be fixed, too. I've pushed an updated revision that also works with spaces. `gitk -- ".\a b*"` will show commits with `a b.txt`.

Reply:
My initial reaction was: why does this only affect backslashes? But after you changed the code, I thought about it a bit more and started to suspect that this need to escape might be unintended. The code to read pathspecs from `stdin` is very clear: https://github.com/git/git/blob/v2.23.0/revision.c#L1914-L1919. As you can see, it reads lines from `stdin`, separated by the Line Feed character, and adds them verbatim to the array of pathspecs. No unescaping, no Bash in the way, so where does this need to escape come from?

Having slept over it, I am even more convinced that something bad is going on. I could imagine, for example, that this escaping is _not_ what we want on Linux.

Do you have any way to test `gitk` on Linux?

Reply:
> so where does this need to escape come from?
My tcl knowledge is limited, but it seems to come from how tcl's `open` is invoked. Running in tclsh:

```
% puts [read [open [concat | dump-stdin.exe "<<foo\\bar"] r] 100]
foar
% puts [read [open [concat | dump-stdin.exe "<<foo\\\\bar"] r] 100]
foo\bar
% puts [read [open |[list dump-stdin.exe "<<foo\\bar"] r] 100]
foo\bar
%
```

Reply:
> Do you have any way to test `gitk` on Linux?

Just tested on Linux and escaping is also required. `gitk 'foo\\*'` will not find the file `foo\bar`.

Reply:
Okay, sorry for the long delay in merging!